### PR TITLE
Fix eval_pv for old compilers

### PR DESCRIPTION
**Pull request by pali**
* Backport `PERL_STATIC_INLINE` macro
* Fix `D_PPP_CROAK_IF_ERROR` when compiler does not support gcc brace groups

Fixes: #216

### CODE REVIEW

**Comment by khwilliamson** [comment]
If this is called multiple times in the same compilation unit, it would be defining the same function multiple times.  I think that is illegal, but maybe no.

**Comment by pali** [comment]
File `call` is included in file `ppport.h` only once and `ppport.h is guarded by `#ifdef` to be processed only once when included in application compilation unit more times. So I think it should be safe.

File: parts/inc/call [hunk 1]
```diff
@@ -51,7 +51,8 @@ __UNDEFINED__ PERL_LOADMOD_IMPORT_OPS   0x4
 #if defined(PERL_USE_GCC_BRACE_GROUPS)
 # define D_PPP_CROAK_IF_ERROR(cond) ({ SV *_errsv; ((cond) && (_errsv = ERRSV) && (SvROK(_errsv) || SvTRUE(_errsv)) && (croak_sv(_errsv), 1)); })
 #else
-# define D_PPP_CROAK_IF_ERROR(cond) ((cond) && (SvROK(ERRSV) || SvTRUE(ERRSV)) && (croak_sv(ERRSV), 1))
+  PERL_STATIC_INLINE void D_PPP_CROAK_IF_ERROR(int cond) { dTHX; SV *errsv; if (!cond) return; errsv = ERRSV; if (SvROK(errsv) || SvTRUE(errsv)) croak_sv(errsv); }
```
Comment (atoomic) [inline parts/inc/call:54]: would adding an `undef D_PPP_CROAK_IF_ERROR` add some extra safety?

**Review by atoomic — APPROVED** [review]

**Comment by atoomic** [comment]
@khwilliamson is going to run some additional extensive tests using this patch before we merge and release it in the next days.

Thanks everyone

**Comment by khwilliamson** [comment]
On 2/10/22 14:55, ℕicolas ℝ. wrote:
> @khwilliamson <https://github.com/khwilliamson> is going to run some 
> additional extensive tests using this patch before we merge and release 
> it in the next days.
> 
> Thanks everyone
> 
> —

And those tests involved making sure that we don't use brace groups at 
all, and the process stopped (recent to older) at 5.9.4 with this:

'In file included from RealPPPort.xs:146:0:
',
                         'RealPPPort.xs: In function 
‘XS_Devel__PPPort_sv_len_utf8’:
',
                         'ppport.h:5926:25: error: expected expression 
before ‘do’
',
                         ' #    define STMT_START  do
',
                         '                         ^
',
 
'/home/khw/devel/lib/perl5/5.9.3/x86_64-linux/CORE/sv.h:1505:23: note: 
in expansion of macro ‘STMT_START’
',
                         ' #define SvGETMAGIC(x) STMT_START { if 
(SvGMAGICAL(x)) mg_get(x); } STMT_END
',
                         '                       ^
',
                         'ppport.h:11071:46: note: in expansion of macro 
‘SvGETMAGIC’
',
                         ' #      define sv_len_utf8(sv) (PL_Sv = (sv), 
SvGETMAGIC(PL_Sv), sv_len_utf8_nomg(PL_Sv))
',
                         '                                              ^
',
                         'RealPPPort.xs:3780:26: note: in expansion of 
macro ‘sv_len_utf8’
',
                         '                 RETVAL = sv_len_utf8(sv);
',

**Comment by atoomic** [comment]
This is fixed by #218 we do not need this PR

**Pull request closed without merging**